vargrid.go

```diff
@@ -697,37 +697,36 @@ func (config *VarGridConfig) createCell(data *CTMData, pop *Population, popIndic
 
 // loadPopMortalityRate calculates the population and baseline mortality rate for this cell.
 func (c *Cell) loadPopMortalityRate(config *VarGridConfig, mortRates *MortalityRates, mortIndices MortIndices, pop *Population, popIndices PopIndices) {
-	for _, mInterface := range mortRates.tree.SearchIntersect(c.Bounds()) {
-		m := mInterface.(*mortality)
-		mIntersection := c.Intersection(m)
-		mAreaIntersect := mIntersection.Area()
-		if mAreaIntersect == 0 {
+	for _, pInterface := range pop.tree.SearchIntersect(c.Bounds()) {
+		p := pInterface.(*population)
+		pIntersection := c.Intersection(p)
+		pAreaIntersect := pIntersection.Area()
+		if pAreaIntersect == 0 {
 			continue
 		}
-		for _, pInterface := range pop.tree.SearchIntersect(mIntersection.Bounds()) {
-			p := pInterface.(*population)
-			pIntersection := c.Intersection(p)
```

Comment:
@ctessum - The main problem was on this line. `pIntersection := c.Intersection(p)` should have been `pIntersection := mIntersection.Intersection(p)`

This was just a typo on my part, but it resulted in the same population getting counted multiple times. It was a pretty obvious error once I looked at some InMAP output because the population counts were nearly a factor of ten too large. It makes me nervous though that this was not caught by any of the tests. I am going to look into that tomorrow.

When looking at the output of the InMAP runs, I also noticed that the total population for the whole country was about 200,000 smaller than in previous runs. I found out that the polygons in the population shapefile (block group scale) extend slightly further out along all borders of the US than the polygons in the mortality rate shapefile (county scale). This was leading to small losses of population in all border cells during the intersection. I realized that I could quickly fix this by just switching the order of the intersections in the `loadPopMortalityRate` function. That is the second change I made. After making this change, the population counts matched the ones I had based on previous versions of InMAP. 200,000 people is a pretty small fraction of 300,000,000 and there is no guarantee that the relative position of population/mortality rate polygon borders will be the same in different analyses, but in general it seems safer to intersect the data that is most likely to be at a finer scale first.